### PR TITLE
ReplicationDestination/Restic Repo for DMR

### PR DESCRIPTION
**Pull request by eemcmullan**

### CODE REVIEW

File: controllers/replicationdestination.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *DataMoverRestoreReconciler) CreateReplicationDestination(log logr.Logger) (bool, error) {
+
+	// get datamoverrestore from cluster
+	dmr := pvcv1alpha1.DataMoverRestore{}
+	if err := r.Get(r.Context, r.NamespacedName, &dmr); err != nil {
```
Comment (savitharaghunathan) [inline controllers/replicationdestination.go:20]: DMR CR will be created in the app ns, right? in that case should we change `r.NamespacedName` -> `r.req.NamespacedName` ?
cc: @shubham-pampattiwar

File: controllers/replicationdestination.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		r.Log.Error(err, "unable to fetch DataMoverRestore CR")
+		return false, err
+	}
+
+	// get datamoverbackup from cluster
+	// TODO: name
+	dmb := pvcv1alpha1.DataMoverBackup{}
```
Comment (savitharaghunathan) [inline controllers/replicationdestination.go:27]: Are we restoring DMB CR? My understanding is that we create a DMR CR (with all the values from DMB CR) for every DMB CR that we encounter in velero backup and ignore restoring DMB CR. Please correct me if I am missing something

File: controllers/replicationdestination.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	// get restic secret created by controller
+	resticSecretName := fmt.Sprintf("%s-secret", dmr.Name)
+	resticSecret := corev1.Secret{}
+	if err := r.Get(r.Context, types.NamespacedName{Namespace: r.NamespacedName.Namespace, Name: resticSecretName}, &resticSecret); err != nil {
+		r.Log.Error(err, "unable to fetch Restic Secret")
+		return err
+	}
+
+	stringCapacity := dmb.Status.SourcePVCData.Size
```
Comment (savitharaghunathan) [inline controllers/replicationdestination.go:70]: Should we populate this field from dmr instead of dmb?

File: controllers/restic.go [hunk 4]
```diff
@@ -137,3 +137,89 @@ func (r *DataMoverBackupReconciler) buildResticSecret(secret *corev1.Secret, dmb
...
 	return nil
 }
+
+// TODO: move these 2 functions to a common.go and check for DMB or DMR being used
+func (r *DataMoverRestoreReconciler) CreateDMRResticSecret(log logr.Logger) (bool, error) {
+
+	// get datamoverbackup from cluster
+	// TODO: name
+	dmb := pvcv1alpha1.DataMoverBackup{}
+	if err := r.Get(r.Context, types.NamespacedName{Name: "datamoverbackup-sample", Namespace: r.NamespacedName.Namespace}, &dmb); err != nil {
```
Comment (savitharaghunathan) [inline controllers/restic.go:147]: nit: remove hard-code name value

File: controllers/replicationdestination.go (hunk 3 above)
Comment (eemcmullan) [reply controllers/replicationdestination.go:70]: Would it be a good idea to put this on DMR?

File: controllers/restic.go (hunk 4 above)
Comment (eemcmullan) [reply controllers/restic.go:147]: Same as above on waiting until DMB is in backup resources.

File: controllers/replicationdestination.go (hunk 2 above)
Comment (eemcmullan) [reply controllers/replicationdestination.go:27]: Just going to leave this in for now until we are able to get DMB from backup.

File: controllers/replicationdestination.go (hunk 3 above)
Comment (savitharaghunathan) [reply controllers/replicationdestination.go:70]: My understanding was DMB CR will not be restored. So when we create DMR CR, we will add all the fields from DMB that we need. 
Would DMR controller know the correct DMB CR to pick?

Comment (eemcmullan) [reply controllers/replicationdestination.go:70]: I gotcha. I guess this is something we will need to hash out once we are able to test getting DMB from backup? In the meantime, okay to leave as is?

Comment (savitharaghunathan) [reply controllers/replicationdestination.go:70]: Yes. May be add a comment/todo so that we dont forget and its good to merge :) Thanks Emily :)

**Review by savitharaghunathan — APPROVED** [review]
lgtm!

Feel free to merge this when you are ready :)

**Pull request merged**